openai.el

```diff
@@ -57,11 +57,36 @@
   (when openai--show-log
     (apply 'message fmt args)))
 
+(defun openai-key--auth-source ()
```

Comment:
I guess this should be public and autoload? Can you rename this to public function without the extra `-`? And add autoload.

```diff
+ ;;;###autoload
- (defun openai-key--auth-source ()
+ (defun openai-key-auth-source ()
```

Reply:
I wondered about whether this should just be documented in the doc string rather than provided. wdyt?

Reply:
I prefer both. Generally, the docstring should give more information regarding the implementation (to the programmer). And documentation in README should just tell the user how to use it. I don't think people look into the code that often. 🤔 

BTW, please ignore the autoload part; I was overthinking it. 😅